frontend/src/pages/NotesHistory.jsx

```
@@ -0,0 +1,417 @@
+        alert('Please log in to download notes');
+        return;
+      }
+
+      // Track download
+      await axios.put(`${API_BASE_URL}/notes/${noteId}/download`, {}, {
+        headers: { Authorization: `Bearer ${token}` }
+      });
+
+      // Find the note
+      const note = [...uploadedNotes, ...downloadedNotes].find(n => n._id === noteId);
+      if (note && note.fileUrl) {
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)